go.mod

```diff
@@ -3,16 +3,28 @@ module github.com/edgexfoundry/go-mod-configuration/v3
 go 1.21
 
 require (
+	github.com/edgexfoundry/go-mod-messaging/v3 v3.1.0
```

Comment:
This creates an implicit dependency on Core Contracts, since go-mod-messaging already has a dependency on Core Contracts. So we might as we'll have this module have a direct dependency on Core Contracts for the constants, DTOs, etc.

Reply:
@lenny-intel about this, if we don't include go-mod-messaging, we have to implement all kinds of message bus here, including redis pub/sub, mqtt, and nats.
we need it to allow the configuration change notice go through the message bus.
or, we have to move the runtime configuration change logic to go-mod-bootstrap

Reply:
Per previous discussion, we are adding go-mod-messaging and go-mod-core-contracts as dependencies.